### ISSUE TITLE
Deployment of Kubernetes job doesn't wait for `Completed` status

### ISSUE THREAD

**Issue by DzeryCZ** [hashicorp/terraform#21992]
Hi there,

I'm doing some experiments with `Kubernetes` provider and I spotted an issue with resource `kubernetes_job`.

I would like to run database migrations via K8s `job` before `deployment` of an application.
So I would like to have the deployment process like this:
- Run Migration Job which takes a minute (check the code)
- When the `job` is marked as `Completed`, create a `deployment`
- Create/Update `service`

My terraform config is:
```
provider "kubernetes" {}

resource "kubernetes_job" "migration" {
  metadata {
    name = "migration"
  }
  spec {
    template {
      metadata {}
      spec {
        container {
          name    = "migration"
          image   = "alpine:latest"
          command = ["sleep", "60"] 
        }
        restart_policy = "Never"
      }
    }
    backoff_limit = 4
    completions = 1
  }
}

resource "kubernetes_deployment" "app" {
  metadata {
    name = "app"
    labels = {
      app = "web"
    }
  }
  spec {
    replicas = 2
    selector {
      match_labels = {
        app = "web"
      }
    }
    template {
      metadata {
        labels = {
          app = "web"
        }
      }
      spec {
        container {
          image = "nginx:1.7.8"
          name  = "example"
          liveness_probe {
            http_get {
              path = "/nginx_status"
              port = 80
            }
            initial_delay_seconds = 15
            period_seconds        = 15
          }
        }
      }
    }
  }
  depends_on = [
      "kubernetes_job.migration"
  ]
}

resource "kubernetes_service" "app" {
  metadata {
    name = "web"
  }
  spec {
    selector = {
      app = "${kubernetes_deployment.app.metadata.0.labels.app}"
    }
    port {
      port        = 80
      target_port = 80
    }
    type = "NodePort"
  }
}
```

When I run `terraform apply` it does:

```
kubernetes_job.migration: Creating...
kubernetes_job.migration: Creation complete after 3s [id=default/migration]
kubernetes_deployment.app: Creating...
kubernetes_deployment.app: Creation complete after 4s [id=default/app]
kubernetes_service.app: Creating...
kubernetes_service.app: Creation complete after 4s [id=default/web]
```

In other words, terraform does not wait until the `job` is marked as `Completed`.

Is it possible to achieve that via any argument? Or is it a bug in terraform?

Many thanks!
